scipy/_lib/tests/test_import_cycles.py

```diff
@@ -1,53 +1,12 @@
 import sys
 import subprocess
 
-
-MODULES = [
-    "scipy.cluster",
-    "scipy.cluster.vq",
-    "scipy.cluster.hierarchy",
-    "scipy.constants",
-    "scipy.fft",
-    "scipy.fftpack",
-    "scipy.fftpack.convolve",
-    "scipy.integrate",
-    "scipy.interpolate",
-    "scipy.io",
-    "scipy.io.arff",
-    "scipy.io.harwell_boeing",
-    "scipy.io.idl",
-    "scipy.io.matlab",
-    "scipy.io.netcdf",
-    "scipy.io.wavfile",
-    "scipy.linalg",
-    "scipy.linalg.blas",
-    "scipy.linalg.cython_blas",
-    "scipy.linalg.lapack",
-    "scipy.linalg.cython_lapack",
-    "scipy.linalg.interpolative",
-    "scipy.misc",
-    "scipy.ndimage",
-    "scipy.odr",
-    "scipy.optimize",
-    "scipy.signal",
-    "scipy.signal.windows",
-    "scipy.sparse",
-    "scipy.sparse.linalg",
-    "scipy.sparse.csgraph",
-    "scipy.spatial",
-    "scipy.spatial.distance",
-    "scipy.special",
-    "scipy.stats",
-    "scipy.stats.distributions",
-    "scipy.stats.mstats",
-    "scipy.stats.contingency"
-]
-
+from .test_public_api import PUBLIC_MODULES, PRIVATE_BUT_PRESENT_MODULES
 
 def test_modules_importable():
     # Regression test for gh-6793.
     # Check that all modules are importable in a new Python process.
     # This is not necessarily true if there are import cycles present.
-    for module in MODULES:
+    for module in PUBLIC_MODULES + PRIVATE_BUT_PRESENT_MODULES:
```

Comment:
I am not sure here, I would just test the public API. IMO it's ok if we accidentally break things on the private side and would force people to move away from doing such things.

Reply:
I tested this locally, it does past but it is very slow. Adding the private modules makes this test take ~4x longer.

Without private modules
`python dev.py test --tests scipy._lib.tests.test_import_cycles  17.32s user 57.71s system 848% cpu 8.838 total`
With private modules
`python dev.py test --tests scipy._lib.tests.test_import_cycles  63.69s user 227.98s system 1008% cpu 28.916 total`

Imagine this could probably be made faster if all imports were checked in one subproccess call rather than one for each import?

Reply:
I think I understand what’s happening. This test really wants to be a multiprocessing pool, but for the sake of simplicity I think it might work to break it up into two separate tests for the public and non-public modules. Let me try that out.

Reply:
Again, not sure we want to test private modules at all.

Reply:
Up to the maintainers, though IMO even if those modules are not stable API they should still be stable from an import perspective.

Reply:
Everything is a maintainability tradeoff. If the test passes with all private modules, we could add it not, but I would consider this more as informative than require this to be green. A compromise could be to mark the test as `xfail` and a final decision would be taken at release time when we review the failed tests.